content/posts/5-grzechow-juniora.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: '5 "grzechów" juniora'
+category: 'Praca'
+publishedAt: '20-12-2021'
+isPublished: false
+popular: false
+image: '/images/antywzorce-css/thumbnail.png'
+excerpt: 'Zaczynanie swojej przygody jako programista rzadko kiedy jest bezstresowe. Nagle trafia się do projektu bez jakiegokolwiek doświadczenia...'
+---
+
+Zaczynanie swojej przygody jako programista rzadko kiedy jest bezstresowe.
+Nagle trafia się do projektu bez jakiegokolwiek doświadczenia, trzeba zdobyć teraz wiedzę domenową na temat projektu, dowiedzieć się kto jest kim w projekcie itp.
+Dlatego by właśnie zminimalizować mozliwe zródła stresu, opiszę jakich rzeczy ( czyli tytułowych "grzechów" ) nie robić.
+No to lecimy!
+
+## Duze samozaparcie w robieniu tasków samemu
+
+Zacznijmy od czegoś według mnie najczęściej spotykanego, czyli próby "pokazania się" poprzez
+robienie tasków samemu. Choćby miało to zająć miesiać to junior chce udowodnić, ze on potrafi! No
+i zazwyczaj kończy się to wręcz przeciwnie, co daily dostajesz pytanie czy juz konczysz tego
+taska, czy moze potrzebujesz pomocy, a ty idąc dalej w zaparte mówisz, ze wszystko jest super
+jednocześnie niepotrzebnie zwiększając swój poziom stresu. Zapamiętaj, nie ma nic złego w
+pytaniu o pomoc czy o pair programming! To nie tylko pomoze ci szybciej rozwiazac twojego taska,
+ale przede wszystkim zwiększy poziom twojej wiedzy, którą zdobędziesz od swojego zespołu.
+
+## Próby pokazania swojej wiedzy na temat róznych bibliotek/frameworków
+
+Jedną z paru rzeczy, które odrózniają juz bardziej doświadczonego programiste od juniora jest
+pewne wyczucie, kiedy jakiś task wymaga biblioteki zewnętrznej bo implementacja i pokrycie
+wszystkich case'ow zajmie zbyt duzo czasu. Niestety tego brakuje wielu juniorom i bardzo łatwo
+jest to zauwazyć juz w repozytoriach takich osób. Projekt w React ToDo-Lista, a w paczkach
+doinstalowane Redux, Redux-Saga, Redux Observable, RxJS, które ostatecznie są uzyte do trzymania
+tasków. No cóz po prostu overengineering pełną gęba, ale za to jakie to uczucie, gdy mozna
+napisać, ze uzyło sie X popularnej technologii u siebie w projekcie! Pamiętaj, biblioteki racja są
```

Comment:
IMO trochę taki atak na juniorów, napisałbym to łagodniej i nie najeżdżał, bo tak to tera brzmi